framework/rest/Serializer.php

```diff
@@ -171,7 +172,11 @@ protected function getRequestedFields()
      */
     protected function serializeDataProvider($dataProvider)
     {
-        $models = $this->serializeModels($dataProvider->getModels());
+        $models = $dataProvider->getModels();
+        if ($dataProvider instanceof ArrayDataProvider) {
```

Comment:
Do we really need this check? I think this bug may appear in any kind of `DataProvider`, not only `ArrayDataProvider`